graylog2-server/src/main/java/org/graylog2/plugin/inputs/Extractor.java

```diff
@@ -159,115 +164,122 @@ public Extractor(MetricRegistry metricRegistry,
-        this.converterTimerName = name(getClass(), getType().toString().toLowerCase(Locale.ENGLISH), getId(), "converterExecutionTime");
+        final String metricsPrefix = name(getClass(), getType().toString().toLowerCase(Locale.ENGLISH), getId());
+        this.conditionHitsCounterName = name(metricsPrefix, "conditionHits");
+        this.conditionMissesCounterName = name(metricsPrefix, "conditionMisses");
+        this.conditionTimerName = name(metricsPrefix, "conditionTime");
+        this.executionTimerName = name(metricsPrefix, "executionTime");
+        this.converterTimerName = name(metricsPrefix, "converterExecutionTime");
+        this.completeTimerName = name(metricsPrefix, "completeExecutionTime");
+        this.conditionHitsCounter = metricRegistry.counter(conditionHitsCounterName);
+        this.conditionMissesCounter = metricRegistry.counter(conditionMissesCounterName);
     }
 
     public void runExtractor(Message msg) {
-        // We can only work on Strings.
-        if (!(msg.getField(sourceField) instanceof String)) {
-            return;
-        }
+        try(final Timer.Context completeTimer = metricRegistry.timer(getCompleteTimerName()).time()) {
+            final String field;
+            try (final Timer.Context conditionTimer = metricRegistry.timer(getConditionTimerName()).time()) {
+                // We can only work on Strings.
+                if (!(msg.getField(sourceField) instanceof String)) {
+                    conditionMissesCounter.inc();
+                    return;
+                }
 
-        final String field = (String) msg.getField(sourceField);
+                field = (String) msg.getField(sourceField);
 
-        // Decide if to extract at all.
-        if (conditionType.equals(ConditionType.STRING)) {
-            if (!field.contains(conditionValue)) {
-                return;
-            }
-        } else if (conditionType.equals(ConditionType.REGEX)) {
-            if (!regexConditionPattern.matcher(field).find()) {
-                return;
+                // Decide if to extract at all.
+                if (conditionType.equals(ConditionType.STRING)) {
+                    if (field.contains(conditionValue)) {
+                        conditionHitsCounter.inc();
+                    } else {
+                        conditionMissesCounter.inc();
+                        return;
+                    }
+                } else if (conditionType.equals(ConditionType.REGEX)) {
+                    if (regexConditionPattern.matcher(field).find()) {
+                        conditionHitsCounter.inc();
+                    } else {
+                        conditionMissesCounter.inc();
+                        return;
+                    }
+                }
             }
-        }
 
-        final Timer.Context timerContext = metricRegistry.timer(getTotalTimerName()).time();
+            try (final Timer.Context executionTimer = metricRegistry.timer(getExecutionTimerName()).time()) {
```

Comment:
nitpick: I prefer to either name these `ignored` or annotate them to avoid the IDE warnings (also elsewhere in this file)